service/javadsl/kafka/server/src/main/scala/com/lightbend/lagom/internal/javadsl/broker/kafka/JavadslRegisterTopicProducers.scala

```diff
@@ -6,33 +6,42 @@ package com.lightbend.lagom.internal.javadsl.broker.kafka
 
 import java.net.URI
 
-import scala.collection.JavaConverters._
-import scala.concurrent.ExecutionContext
-import scala.concurrent.Future
-import org.slf4j.LoggerFactory
-import com.lightbend.lagom.javadsl.api.ServiceInfo
-import com.lightbend.lagom.javadsl.api.ServiceLocator
-import akka.stream.Materializer
-import javax.inject.Inject
+import akka.NotUsed
 import akka.actor.ActorSystem
+import akka.persistence.query.EventEnvelope
 import akka.persistence.query.Offset
+import akka.stream.Materializer
 import akka.stream.scaladsl.Source
+import com.lightbend.lagom.internal.broker.DelegatedTopicProducer
+import com.lightbend.lagom.internal.broker.TaggedInternalTopic
 import com.lightbend.lagom.internal.broker.TaggedOffsetTopicProducer
+import com.lightbend.lagom.internal.broker.kafka.ClassicLagomEventStreamFactory
+import com.lightbend.lagom.internal.broker.kafka.DelegatedEventStreamFactory
+import com.lightbend.lagom.internal.broker.kafka.EventStreamFactory
 import com.lightbend.lagom.internal.broker.kafka.KafkaConfig
 import com.lightbend.lagom.internal.broker.kafka.Producer
-import com.lightbend.lagom.internal.projection.ProjectionRegistry
 import com.lightbend.lagom.internal.javadsl.api.MethodTopicHolder
 import com.lightbend.lagom.internal.javadsl.api.broker.TopicFactory
 import com.lightbend.lagom.internal.javadsl.persistence.OffsetAdapter
 import com.lightbend.lagom.internal.javadsl.server.ResolvedServices
+import com.lightbend.lagom.internal.projection.ProjectionRegistry
 import com.lightbend.lagom.javadsl.api.Descriptor.TopicCall
+import com.lightbend.lagom.javadsl.api.ServiceInfo
+import com.lightbend.lagom.javadsl.api.ServiceLocator
 import com.lightbend.lagom.javadsl.api.broker.kafka.KafkaProperties
+import com.lightbend.lagom.javadsl.persistence.AggregateEvent
+import com.lightbend.lagom.javadsl.persistence.AggregateEventTag
 import com.lightbend.lagom.spi.persistence.OffsetStore
+import javax.inject.Inject
+import org.slf4j.LoggerFactory
 
+import scala.collection.JavaConverters._
 import scala.collection.immutable
 import scala.compat.java8.FutureConverters._
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
 
-class JavadslRegisterTopicProducers @Inject() (
+class JavadslRegisterTopicProducers[BrokerMessage, Event <: AggregateEvent[Event]] @Inject() (
```

Comment:
did a bit of a cleanup trying to enforce types (avoid `AnyRef`'s and `[_]` as much as possible).